core/src/main/java/io/grpc/Deadline.java

```diff
@@ -139,4 +139,9 @@ public long timeRemaining(TimeUnit unit) {
     Preconditions.checkNotNull(scheduler, "scheduler");
     return scheduler.schedule(task, deadlineNanos - System.nanoTime(), TimeUnit.NANOSECONDS);
   }
+
+  @Override
+  public String toString() {
+    return String.valueOf(deadlineNanos);
```

Comment:
Doing "X ns from now" may not be a bad idea. Just the nanos is useless without a reference to the clock.